### PR TITLE
feat: add main function

### LINKED ISSUE #62: Add `main` function that reads `Input` from stdin and prints the value of `LAUNCH`

### PR DESCRIPTION
**Pull request by mantaur**
fixes: #62 
Adds main function that can read from standard input or alternatively a specified input file.
Includes fix:s so that main can run underlying functions without naming conflicts.
Modified build.gradle so that a task `run1` can be run with specified args to location of test file.

### CODE REVIEW

File: src/main/java/decide/Main.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        // preliminary unlocking matrix (pum)
+        PreliminaryUnlockingMatrix pum = new PreliminaryUnlockingMatrix(input, cmv);
+
+        // Use the preliminary unlocking vector (puv) of the input along with the pum to calculate
+        // the final unlocking vector (fuv)
+        FinalUnlockingVector fuv = new FinalUnlockingVector(input.puv, pum);
+
+        // Finally calculate the output launch boolean and print the result
+        Output output = new Output(fuv);
+        System.out.println(output.launch);
```
Comment (nolanderc) [inline src/main/java/decide/Main.java]: According to the specification we should either output `YES` or `NO`. Might be nicer than `true`/`false`.

File: src/test/java/decide/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    @Test
+    void main_test_with_args() {
+        Main.main(new String[] {"./in/input1.txt"});
+    }
+
+    // Must redirect System.in or will read empty input and crash
+    // @Test
+    // void main_test_stdin() {
+    //     Main.main(new String[0]); 
+    // }
```
Comment (nolanderc) [inline src/test/java/decide/MainTest.java]: Is this comment meant to be here?

File: build.gradle [hunk 3]
```diff
@@ -10,6 +10,13 @@ repositories {
 dependencies {
     testImplementation('org.junit.jupiter:junit-jupiter:5.8.2')
 }
+
+// Pass args to arg1 like so: `./gradlew run1 -Parg1="argument value goes here" `
+task run1(type: JavaExec) {
+    mainClassName = "decide.Main"
+    classpath = sourceSets.main.runtimeClasspath
+    // args = ["$arg1"]    // Uncomment this line if you wish to pass args on cmd line
+}
```
Comment (nolanderc) [inline build.gradle]: I don't think we need an entirely new task for this. You can instead use `gradle run --args="<path to file>"`

**Review by nolanderc — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (mantaur) [reply build.gradle]: ahh nice, I used the first best solution I found for passing an arg to mainClassName main function, but this works like a charm. Will revert these changes

File: src/test/java/decide/MainTest.java (hunk 2 above)
Comment (mantaur) [reply src/test/java/decide/MainTest.java]: It was indeed, left mostly as a reminder for how if no argument with path to a file is used and the main function assumes System.in, this stream must be redirected in advance or will cause a crash.

File: src/main/java/decide/Main.java (hunk 1 above)
Comment (mantaur) [reply src/main/java/decide/Main.java]: True, will update this

File: src/test/java/decide/MainTest.java (hunk 2 above)
Comment (nolanderc) [reply src/test/java/decide/MainTest.java]: In that case, it might be better to catch the exception in order to make the intent explicit.

Comment (mantaur) [reply src/test/java/decide/MainTest.java]: I see a quick and a thorough solution to this. The quick being just catching whatever exception is thrown at the main level. Or the thorough solution of reworking the input class extensively with some custom input exception and throwing it where ever it previously threw exceptions.

Comment (nolanderc) [reply src/test/java/decide/MainTest.java]: Yeah, a custom exception is probably best, but the change in ead549194548139219c8b8833c9747ec49b34fea is probably good enough.

**Review by nolanderc — APPROVED** [review]

**Pull request merged**